students/smitco/lesson03/strformat_lab.py

```diff
@@ -0,0 +1,75 @@
+#lesson 03 string formatting
+
+#task 1- take tuple and print formatted string "file_002: 123.46, 1.00e+04, 1.23e+04"
+#use {:.2f} for float with 2 decimal places
+#use {:0>3d} for padding on the left with zeroes for 3 numbers width
+#use {:.2e} for exponential with 2 decimal places
+tupelo = (2, 123.4567, 10000, 12345.67)
+filename = "file_{:0>3d}: {:.2f}, {:.2e}, {:.2e}"
+print(filename.format(*tupelo))
+print()
+print()
+
+#task 2- repeat task 1 with an alternate type of format string
+filename = f"file_{tupelo[0]:0>3d}: {tupelo[1]:.2f}, {tupelo[2]:.2e}, {tupelo[3]:.2e}"
+print(filename)
+print()
+print()
+
+#task 3- dynamically rewrite "the 3 numbers are: {:d}, {:d}, {:d}".format(1,2,3) 
+testtup = (1, 2, 3, 4, 5, 6)
+testtup2 = (4, 7, 5)
+testtup3 = (6, 19, 22, 34)
+def strform(tup):
+    l = len(tup)
+    lst = ""
```

Comment:
It's a good idea to always use meaningful variable names, and in the case of `l`, you probably don't need to assign that to a variable at all (just call `len` where you need it)